### PR TITLE
Toggle unit price feature to individual users by email

### LINKED ISSUE #6870: Toggle unit prices feature to individual users
## What?
We must activate unit price feature to individual users: list of email users in a `env` variable, such as `UNIT_PRICE_BETA_TESTERS`.

## Expected Behavior
Unit price must be shown in the shopfront (or elsewhere) only for users who have email in the list.

## Context
Context: https://github.com/openfoodfoundation/openfoodnetwork/pull/6851#discussion_r575296244

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?
Closes #6870

This PR make it possible to activate unit price feature for connected user, one by one, with their email. This is done by filling the `env` variable  `BETA_TESTERS_FOR_UNIT_PRICE`, which is basically a list a email separated by comma :
```
BETA_TESTERS_FOR_UNIT_PRICE: ofn@example.com
```

This PR does't change the previous behavior, that is to say: 
_The unit price feature is activated on `staging` and `development` environment._ 

#### What should we test?
The feature is still active for `staging` and `development` environment, so testing is, to me, almost impossible.
But maybe @sauloperez has any idea?

#### Release notes
Enable the activation for unit price feature to individual user.

Changelog Category: Technical changes

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]
I think this looks fine as it is, at least in terms of doing what it's supposed to do. So if we want to push forward on it I think that's fine. 

However, I think this could quickly get really messy if we're maintaining a separate list of emails that are enabled for each feature. I'll defer to @sauloperez's judgment on whether we want to use this approach, or if https://github.com/jnunemaker/flipper is looking more and more appealing....

**Comment by sauloperez** [comment]
I also share that concern @andrewpbrett. However, with @jibees we saw that Flipper won't solve our problems it's just an implementation of feature toggles (that might be better than ours) and as such it doesn't take any business decisions for us. I'm scared but I see that sooner or later we'll need to have a user-set per feature.

I feel like we need to stop and assess where we want to go with this and come up with some sort of plan. Fancy a quick chat next week? can we wait a few more days @jibees ?

**Comment by andrewpbrett** [comment]
Yeah, chat next week sounds great 👍

**Review by mkllnk — APPROVED** [review]
Good work. This can go ahead and we can make a decision about other feature toggles later.

I find a gem like flipper very appealing. It solves it for us and is very flexible. I like that we can change the affected users on the server. We could even offer people to opt in voluntarily. The downside of flipper is that it seems quite big and it won't be as fast as our own in-memory version.

**Comment by sauloperez** [comment]
valid points @mkllnk however, I believe Flipper also has an in-memory adapter.

**Comment by mkllnk** [comment]
Yes, but I would guess that it's not the same. We store data in environment variables and then serve it readonly from memory. I think that flipper allows to modify data, stores it in memory and we lose it on a restart. So I would use the ActiveRecord adapter for persistence. Flipper claims to be very efficient. It might not be a problem at all. Even if a feature flag triggers a handful of SQL queries on each request that's nothing compared to the hundreds of queries triggered by callbacks.

**Comment by jibees** [comment]
Just to have in mind another need: #6953 // Activate feature for unlogged user

**Comment by sauloperez** [comment]
We decided that we'll investigate https://github.com/jnunemaker/flipper/blob/master/docs/ui/README.md to see if that's a better approach. If it works as expected it'd enable product people to define groups of users without needing a dev or a server provisioning.

Still, we think defining the groups plus adding/removing users for each instance should be a core team responsibility. Ideally, a product/tech lead one.

**Comment by jibees** [comment]
@sauloperez I'll take it ;)

**Comment by jibees** [comment]
I've started a document : https://docs.google.com/document/d/1P3j2i9T1NVx2dguzTofrOmLzMG4eIR3KcN5RVdDcLcY/edit Feel free to edit/comment/... ! 🙏

**Comment by sauloperez** [comment]
Alright, I went through your doc @jibees and I think it'll be great to try it. Above all, it brings more visibility to product people and testers and so, it'll make it more clear for them who's got what enabled and reduce the cost of enabling/disabling features.

It could also make it a bit more maintainable in the code base as well. 

Thoughts @andrewpbrett @mkllnk ? I'd say let's try it for unit prices.

**Comment by mkllnk** [comment]
Yes, I'm happy to try flipper.

**Comment by jibees** [comment]
I created a new issue #7193 
Thus, I propose to delete this PR (and associated issue). Wdyt? @sauloperez

**Pull request closed without merging**
